pimd/pim_nht.c

```diff
@@ -75,10 +76,7 @@ struct pim_nexthop_cache *pim_nexthop_cache_find(struct pim_instance *pim,
 	struct pim_nexthop_cache *pnc = NULL;
 	struct pim_nexthop_cache lookup;
 
-	lookup.rpf.rpf_addr.family = rpf->rpf_addr.family;
-	lookup.rpf.rpf_addr.prefixlen = rpf->rpf_addr.prefixlen;
-	lookup.rpf.rpf_addr.u.prefix4.s_addr = rpf->rpf_addr.u.prefix4.s_addr;
-
+	prefix_copy(&(lookup.rpf.rpf_addr), &(rpf->rpf_addr));
```

Comment:
`prefix_copy` is only needed when we're mixing `struct prefix_ipv4` `struct prefix_ipv6` etc.

Just `lookup.rpf.rpf_addr = rpf->rpf_addr` should be good here.

(Only single comment, have not reviewed rest of this PR yet.)